File: backend/internal/model/faq.go
package model

import "time"

type FAQ struct {
	ID          string    `json:"id"`
	Slug        string    `json:"slug"`
	Category    string    `json:"category"`
	Question    string    `json:"question"`
	AnswerMD    string    `json:"answer_md"`
	SortOrder   int       `json:"sort_order"`
	IsPublished bool      `json:"is_published"`
	UpdatedBy   *string   `json:"updated_by,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CreateFAQInput struct {
	Slug        string `json:"slug"`
	Category    string `json:"category"`
	Question    string `json:"question"`
	AnswerMD    string `json:"answer_md"`
	SortOrder   int    `json:"sort_order"`
	IsPublished *bool  `json:"is_published"`
}

type UpdateFAQInput struct {
	Slug        *string `json:"slug"`
	Category    *string `json:"category"`
	Question    *string `json:"question"`
	AnswerMD    *string `json:"answer_md"`
	SortOrder   *int    `json:"sort_order"`
	IsPublished *bool   `json:"is_published"`
}

File: frontend/src/api/faq.ts
import { api } from './client'

export interface FAQ {
  id: string
  slug: string
  category: string
  question: string
  answer_md: string
  sort_order: number
  is_published: boolean
  updated_by?: string
  created_at: string
  updated_at: string
}

export interface FAQListResponse {
  items: FAQ[]
}

export const faqApi = {
  list: () => api.get<FAQListResponse>('/faqs'),
  getBySlug: (slug: string) => api.get<FAQ>(`/faqs/${slug}`),
}
